MathlibTest/fun_prop_dev.lean

```diff
@@ -553,3 +554,49 @@ example [Add α] (y : α):
   have : Con f := by fun_prop
   set_option trace.Meta.Tactic.fun_prop true in
   fun_prop
+
+
+
+--- pefromance tests - mainly testing fast failure ---
+------------------------------------------------------
+
+
+section PerformanceTests
+-- set_option trace.Meta.Tactic.fun_prop true
+-- set_option profiler true
+
+variable {R : Type*} [Add R] [∀ n, OfNat R n]
+example (f : R → R) (hf : Con f) :
+    Con (fun x ↦ f (x + 3)) := by fun_prop -- succeeds in 5ms
+example (f : R → R) (hf : Con f) :
+    Con (fun x ↦ (f (x + 3)) + 2) := by fun_prop -- succeeds in 6ms
+example (f : R → R) (hf : Con f) :
+    Con (fun x ↦ (f (x + 3)) + (2 + f (x + 1))) := by fun_prop -- succeeds in 8ms
+example (f : R → R) (hf : Con f) :
+    Con (fun x ↦ (f (x + 3)) + (2 + f (x + 1)) + x) := by fun_prop -- succeeds in 10ms
+example (f : R → R) (hf : Con f) :
+    Con (fun x ↦ (f (x + 3)) + 2 + f (x + 1) + x + 1) := by fun_prop -- succeeds in 11ms
+
+-- this used to fail in exponentially increasing time, up to 6s for the last example
```

Comment:
Could you turn these into actual tests? (e.g. by using `fail_if_success`/`success_if_fail_with_msg` + some low timeout (whose failure is not caught by the `fail_if_success`)